client/app/bundles/course/assessment/submission/components/answers/Programming/ProgrammingFilenameWithDownload.tsx

```diff
@@ -0,0 +1,31 @@
+import { FC } from 'react';
+import { Download } from '@mui/icons-material';
+import { IconButton, Typography } from '@mui/material';
+import { ProgrammingContent } from 'types/course/assessment/submission/answer/programming';
+import { downloadFile } from 'utilities/downloadFile';
+
+interface ProgrammingFileDownloadLinkProps {
+  file: ProgrammingContent;
+}
+
+const ProgrammingFilenameWithDownload: FC<ProgrammingFileDownloadLinkProps> = (
+  props,
+) => {
+  const { file } = props;
+  const filename = file.filename;
+
+  const handleDownload = (): void => {
+    downloadFile('text/plain', file.content, filename);
+  };
+
+  return (
+    <div style={{ display: 'flex', alignItems: 'center', gap: '10px' }}>
+      <Typography variant="body1">{filename}</Typography>
+      <IconButton aria-label="download file" onClick={handleDownload}>
```

Comment:
Current download icon is very large. Filename text is also very large.  
How about a clickable chip similar to this: ![image](https://github.com/user-attachments/assets/007f0e50-91a3-4f2c-aa6b-93ee53242fd1)  
`<Chip label="MyClass.java" size="small" color="primary" variant="outlined" icon={<DownloadIcon />} />`

Reply:
Done